x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -431,29 +467,42 @@ func newMetrics(maxGroups int) *metrics {
 }
 
 type metricsMapEntry struct {
-	transactionAggregationKey
 	transactionMetrics
+	transactionAggregationKey
```

Comment:
This saves 16 bytes of padding. Since we might have up to 1000 or more entries in memory at any time it seemed like a good place to save a few bytes.

Reply:
There's a linter I use called [`fieldalignment`](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.8:go/analysis/passes/fieldalignment/fieldalignment.go). You can easily configure it in `vscode` `gopls` so that it analyzes the struct field alignment and displays a warning.

```json
...
    "gopls": {
        "ui.diagnostic.analyses": {
            "fieldalignment": true
        },
    },
...
```

You'll need to install the tool first but it's easy enough: `go install golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@latest`.